text/0002-sarif-result-schema.md

```diff
@@ -0,0 +1,482 @@
+- Feature Name: Migrating Result Schema to SARIF 
+- Start Date: 2020-09-01
+- RFC PR:
+- Checkup Issue: 
+
+# Summary
+[summary]: #summary
+
+In [a previous RFC](./0001-task-result-schema.md), the need to standardize the output of Checkup's data was discussed. In this RFC, I propose leveraging the [SARIF](https://github.com/microsoft/sarif-tutorials) schema to shape the Checkup output. 
+
+# Motivation
+[motivation]: #motivation
+
+Since the [Task Result Schema RFC](./0001-task-result-schema.md), changes have been implemented to the output that align with the schema described in that RFC, cleaning up the output significantly and adding shared helpers that help task authors format their data properly to fit the schema. SARIF is a well-established schema that "defines a standard format for the output of static analysis tools." There are a ton of tools and plugins available in the open-source community that leverage SARIF - aligning our data structure with this shared schema will allow the data produced by checkup to be consumed by data pipelines that are already consuming SARIF data. For example, there is a [VSCode plugin](https://github.com/microsoft/sarif-vscode-extension) that prettyprints SARIF data, a [react based visualization tool](https://github.com/Microsoft/sarif-web-component), and a package that [converts eslint data to SARIF](https://www.npmjs.com/package/%40microsoft%2Feslint-formatter-sarif
```

Comment:
```suggestion
Since the [Task Result Schema RFC](./0001-task-result-schema.md), changes have been implemented to the output that align with the schema described in that RFC, cleaning up the output significantly and adding shared helpers that help task authors format their data properly to fit the schema. Though defining a refined schema was a useful step in structuring the output, it didn't leverage any standardized schema for static analysis, such as the SERIF format from Microsoft.

SARIF is a well-established schema that "defines a standard format for the output of static analysis tools." There are a multitude of tools and plugins available in the open-source community that leverage SARIF - aligning our data structure with this shared schema will allow the data produced by checkup to be consumed by data pipelines that are already consuming SARIF data. For example, there is a [VSCode plugin](https://github.com/microsoft/sarif-vscode-extension) that prettyprints SARIF data, a [react based visualization tool](https://github.com/Microsoft/sarif-web-component), and a package that [converts eslint data to SARIF](https://www.npmjs.com/package/%40microsoft%2Feslint-formatter-sarif
```

Reply:
accepted suggestion